### ISSUE TITLE
HttpClient scope

### ISSUE THREAD

**Issue by theo-albers** [buckaroo-it/BuckarooSdk_DotNet#35]
The Connection class creates an HttpClient and DelegatingHandler on every request. This is bad practice in a server environment like ASP.NET, see
https://docs.microsoft.com/en-us/dotnet/standard/microservices-architecture/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests. It would be very nice to be able to control the HttpClient or DelegateHandler lifetime as SDK consumer. Right now we can't use the SDK (or we would have to clone and modify the source).

**Comment by buckaroo-it** [comment]
Thank you for your suggestion. I will take a look at this as I agree with your point of view on this matter. I will at least provide you with the intended solution and the timeline to implement the fix. In the coming weeks we are short staffed and a change like this needs to be tested thoroughly. I will come back to you. greetings

**Comment by jonassix** [comment]
Hi,

What is the status of this?

**Comment by stefankip** [comment]
...

**Comment by jonassix** [comment]
> ...

Yeah, they won't respond. I've developed my own implementation faster than they are even able to respond

**Comment by timvandenhof** [comment]
> Yeah, they won't respond. I've developed my own implementation faster than they are even able to respond

Can you open a PR for them so other can profit too?
My current team is implementing Buckaroo at the moment, and this could hinder our performance as mentioned by Theo.

**Comment by jonassix** [comment]
I've written an entire code base from scratch. It's not a fork and it's not open source. 
So won't do a PR.

This library was very handy as a form of documenation. For how authentication works , how the message's are constructed. 
Imo the only "difficult" thing was the authentication. So my advice would be to just build your own implementation.
